### DISCUSSION TITLE
Invoice metadata questions

### DISCUSSION THREAD

**Discussion by Ou7law007** [btcpayserver/btcpayserver#3974] [Technical Support] [answered]
When creating an invoice using the API, there's an option to provide metadata.

https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_CreateInvoice

I thought these were for the devs for internal usage!

Although, it says "You can use this property to store the ID of an **external system**. We allow you to search in the invoice list based on this ID.", but apparently, the orderId field can be accessed by the end user as well as the orderUrl

1. Can the end user access any other metadata fields such as buyerName?
2. Is it a bad idea to store sensitive data in the invoice's metadata?
3. Is there a general way to know if any piece of information is accessible by the end user? The API says: AUTHORIZATIONS:
API_Key (btcpay.store.cancreateinvoice) AND Basic, but this apply to the request itself and not the information it provides. But we see some fields in metadata, for example, are accessible to the end user.

Thank you

**Comment by pavlenex** [comment]
cc @NicolasDorier can you help with this one? Is there something we can do to better document the metadata?

**Comment by bolatovumar** [comment]
> Can the end user access any other metadata fields such as buyerName?

@Ou7law007 Depends on what kind of access the end user has. If the user can see the invoice if the BTCPay admin panel (i.e, if they have the "Can view invoice" permission) or have the API key that has the permission to view invoices then they can see this metadata.

> Is it a bad idea to store sensitive data in the invoice's metadata?

If you are unsure who has access to view the invoices then yes. You should know who has access to your instance of BTCPay though. You can check on the users page to see who has which role and you can check the API keys page to see which API keys exist.

> Is there a general way to know if any piece of information is accessible by the end user? The API says: AUTHORIZATIONS:
API_Key (btcpay.store.cancreateinvoice) AND Basic, but this apply to the request itself and not the information it provides. But we see some fields in metadata, for example, are accessible to the end user.

See the above reply.

**Reply by Ou7law007** [reply]
Nobody else has access to the store (the users tab is simply empty). Obviously, the customers has access to their respective invoices so that they can pay them, and by access I don't mean using an API key but they can simply view the invoice and they'll see the order ID, for example, so I'm not sure if they can (in whatever way) be able see anything else attached to `meta` other than `orderId` which is one of the fields inside of `meta`

**Reply by bolatovumar** [reply, marked as answer]
I don't there is a way anyone can see this metadata then.
